tests/services/application.service.spec.ts

```diff
@@ -1,18 +1,54 @@
 import { ApplicationService } from '../../src/app/core/services/application.service';
+import { GithubService } from '../../src/app/core/services/github.service';
+import { of } from 'rxjs';
 
-const applicationService = new ApplicationService(null);
 const currentVersion = '3.2.5';
 
-describe('Test the ApplicationService', () => {
-    it('Test whether the ApplicationService can detect outdated versions', () => {
-        expect(applicationService.isOutdatedVersion('2.2.5', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.1', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.4', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.4.8.9', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion(currentVersion, currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.5.0', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('3.2.5.5', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('3.2.6', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('10', currentVersion)).toBe(true);
-    });
+class ApplicationServiceStub extends ApplicationService {
+  readonly currentVersion: string;
+  constructor(githubService: GithubService, currentVersion: string) {
+    super(githubService);
+    this.currentVersion = currentVersion;
+  }
+}
+
+describe('Test for ApplicationService#isApplicationOutdated', () => {
```

Comment:
Let's shorten this description to `ApplicationService#isApplicationOutdated` (the `Test for` is redundant)

Reply:
Yu Chen, could you make this final change? Thanks!

Reply:
Will do, was a little preoccupied these 2 days.